corehq/motech/auth.py

```diff
@@ -0,0 +1,38 @@
+import requests
+import re
+from urllib.parse import urljoin
+from requests.exceptions import RequestException
+
+
+class HTTPBearerAuth(requests.auth.AuthBase):
+    def __init__(self, username, plaintext_password):
+        self.username = username
+        self.password = plaintext_password
+
+    def _find_bearer_base(self, r):
+        m = re.compile('https.*/api/v[0-9]+/').match(r.url)
```

Comment:
Yeah, I don't love it. Unfortunately the base is a little bit implicit because as far a I can tell they are using a pseudo-standard, but I think this would match any system using the underlying api library.